trino/auth.py

```diff
@@ -251,13 +281,48 @@ def _get_token(self, token_server, response, **kwargs):
 
         raise exceptions.TrinoAuthError("Exceeded max attempts while getting the token")
 
+    def _get_token_from_cache(self, host):
+        with self._token_lock:
+            if self._keyring is not None:
+                # keyring is installed, so we can reuse the same token over multiple threads
+                try:
+                    return self._keyring.get_password(
+                        f"{host}", "token"
+                    )
+                except self._keyring.errors.NoKeyringError:
+                    pass
+
+            # keyring is not installed, so we fall back to a per connection cache
```

Comment:
per instance of `OAuth2Authentication` right?

Reply:
That's indeed correct, also mentioned in earlier reviews. I will update the comment.